manageiq-operator/api/v1alpha1/helpers/miq-components/kafka/kafka.go

```diff
@@ -0,0 +1,351 @@
+				map[string]interface{}{
+					"resource": map[string]interface{}{
+						"type":        "topic",
+						"name":        "*",
+						"patternType": "literal",
+					},
+					"operations": []string{"All"},
+					"host":       "*",
+				},
+				map[string]interface{}{
+					"resource": map[string]interface{}{
+						"type":        "group",
+						"name":        "*",
+						"patternType": "literal",
+					},
+					"operations": []string{"All"},
+					"host":       "*",
+				},
+			},
+		},
+	}
+}
+
+func KafkaUser(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*unstructured.Unstructured, controllerutil.MutateFn) {
+	kafkaUserCR := &unstructured.Unstructured{}
+
+	kafkaUserCR.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "kafka.strimzi.io",
+		Kind:    "KafkaUser",
+		Version: "v1beta2",
+	})
+	kafkaUserCR.SetName("manageiq-user")
+	kafkaUserCR.SetNamespace(cr.Namespace)
+	kafkaUserCR.SetLabels(map[string]string{"strimzi.io/cluster": "manageiq"})
+
+	kafkaUserSpec := KafkaUserSpec()
+
+	mutateFunc := func() error {
+		if err := controllerutil.SetControllerReference(cr, kafkaUserCR, scheme); err != nil {
+			return err
+		}
+
+		kafkaUserCR.UnstructuredContent()["spec"] = kafkaUserSpec
+
+		return nil
+	}
+
+	return kafkaUserCR, mutateFunc
+}
+
+func KafkaTopicSpec() (map[string]interface{}) {
+	return map[string]interface{}{
+		"partitions": 1,
+		"config": map[string]interface{}{
+			"retention.ms":  7200000,
+			"segment.bytes": 1073741824,
+		},
+	}
+}
+
+func KafkaTopic(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme, topicName string) (*unstructured.Unstructured, controllerutil.MutateFn) {
+	kafkaTopicCR := &unstructured.Unstructured{}
+
+	kafkaTopicCR.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "kafka.strimzi.io",
+		Kind:    "KafkaTopic",
+		Version: "v1beta2",
+	})
+	kafkaTopicCR.SetName(topicName)
+	kafkaTopicCR.SetNamespace(cr.Namespace)
+	kafkaTopicCR.SetLabels(map[string]string{"strimzi.io/cluster": "manageiq"})
+
+	kafkaTopicSpec := KafkaTopicSpec()
+
+	mutateFunc := func() error {
+		if err := controllerutil.SetControllerReference(cr, kafkaTopicCR, scheme); err != nil {
+			return err
+		}
+
+		kafkaTopicCR.UnstructuredContent()["spec"] = kafkaTopicSpec
+
+		return nil
+	}
+
+	return kafkaTopicCR, mutateFunc
+}
+
+func KafkaInstall(cr *miqv1alpha1.ManageIQ, scheme *runtime.Scheme) (*unstructured.Unstructured, controllerutil.MutateFn) {
+	kafkaSubscription := &unstructured.Unstructured{}
+	kafkaSubscription.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "operators.coreos.com",
+		Kind:    "Subscription",
+		Version: "v1alpha1",
+	})
+	kafkaSubscription.SetName("strimzi-kafka-operator")
+	kafkaSubscription.SetNamespace(cr.Namespace)
+
+	kafkaSubscriptionSpec := map[string]interface{}{
+		"channel":         "strimzi-0.35.x",
+		"name":            "strimzi-kafka-operator",
+		"source":          "community-operators",
+		"sourceNamespace": "openshift-marketplace",
+	}
```

Comment:
Fixed the strimzi release to 0.35.1 to match downstream, also to prevent breaking changes that might be introduced in latest releases